builtin/k8s/task.go

```diff
@@ -0,0 +1,271 @@
+
+	// Set an explicit pull policy for this task launching. By default
+	// we use "PullIfNotPresent" unless the image tag is "latest" when we
+	// use "Always".
+	PullPolicy string `hcl:"image_pull_policy,optional"`
+}
+
+func (p *TaskLauncher) Documentation() (*docs.Documentation, error) {
+	doc, err := docs.New(
+		docs.FromConfig(&TaskLauncherConfig{}),
+		docs.FromFunc(p.StartTaskFunc()),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	doc.Description(`
+Launch a Kubernetes pod for on-demand tasks from the Waypoint server.
+
+This will use the standard Kubernetes environment variables to source
+authentication information for Kubernetes. If this is running within Kubernetes
+itself (typical for a Kubernetes-based installation), it will use the pod's
+service account unless other auth is explicitly given. This allows the task
+launcher to work by default.
+`)
+
+	doc.Example(`
+task {
+	use "kubernetes" {}
+}
+`)
+
+	doc.SetField(
+		"kubeconfig",
+		"path to the kubeconfig file to use",
+		docs.Summary("by default uses from current user's home directory"),
+		docs.EnvVar("KUBECONFIG"),
+	)
+
+	doc.SetField(
+		"context",
+		"the kubectl context to use, as defined in the kubeconfig file",
+	)
+
+	doc.SetField(
+		"image_secret",
+		"name of the Kubernetes secrete to use for the image",
+		docs.Summary(
+			"this references an existing secret, waypoint does not create this secret",
+		),
+	)
+
+	doc.SetField(
+		"service_account",
+		"service account name to be added to the application pod",
+		docs.Summary(
+			"service account is the name of the Kubernetes service account to add to the pod.",
+			"This is useful to apply Kubernetes RBAC to the application.",
+		),
+	)
+
+	doc.SetField(
+		"image_pull_policy",
+		"pull policy to use for the task container image",
+	)
+
+	return doc, nil
+}
+
+// TaskLauncher implements Configurable
+func (p *TaskLauncher) Config() (interface{}, error) {
+	return &p.config, nil
+}
+
+// StopTask signals to docker to stop the container created previously
+func (p *TaskLauncher) StopTask(
+	ctx context.Context,
+	log hclog.Logger,
+	ti *TaskInfo,
+) error {
+	// Get our client
+	clientSet, ns, _, err := clientset(p.config.KubeconfigPath, p.config.Context)
+	if err != nil {
+		return err
+	}
+
+	// Get our jobs client
+	jobsClient := clientSet.BatchV1().Jobs(ns)
+	err = jobsClient.Delete(ctx, ti.Id, metav1.DeleteOptions{})
+	if errors.IsNotFound(err) {
+		// If it doesn't exist then that's fine, its already stopped then.
+		err = nil
+	}
+
+	return err
+}
+
+// StartTask creates a docker container for the task.
+func (p *TaskLauncher) StartTask(
+	ctx context.Context,
+	log hclog.Logger,
+	tli *component.TaskLaunchInfo,
+) (*TaskInfo, error) {
+	// Get our client
+	clientSet, ns, _, err := clientset(p.config.KubeconfigPath, p.config.Context)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate an ID for our pod name.
+	id, err := ulid.New(ulid.Now(), rand.Reader)
```

Comment:
We already import `sdk/component`, should we use [`component.Id()`][id] here instead? 


[id]: https://github.com/hashicorp/waypoint-plugin-sdk/blob/f88287ee9c998bdb3bcb626bc1ef9f195a53f043/component/id.go#L14

Reply:
We could, but I think this will change in a bit anyways once I start working on https://github.com/hashicorp/waypoint/issues/2116 so lets leave it for now (this is also how Docker does it).